src/components/scheduling/goals/SchedulingGoalForm.svelte

```diff
@@ -149,8 +190,21 @@
             }
           }
 
+          // Associate new tags with expansion rule
+          const newSchedulingGoalTags: SchedulingGoalTagsInsertInput[] = (goalTags || []).map(({ id: tag_id }) => ({
+            goal_id: goalId as number,
+            tag_id,
+          }));
+          await effects.createSchedulingGoalTags(newSchedulingGoalTags, user);
+
+          // Disassociate old tags from constraint
+          const unusedTags = initialGoalTags
+            .filter(tag => !(goalTags || []).find(t => tag.id === t.id))
+            .map(tag => tag.id);
+          await effects.deleteSchedulingGoalTags(unusedTags, user);
+
           goalModifiedDate = updatedGoal.modified_date;
-          savedGoal = { ...goal };
+          savedGoal = { ...goal, tags: (goalTags || []).map(tag => ({ tag })) };
```

Comment:
Minor. Would `??` make more sense than `||` here since we want to default to `[]` if `goalTags` does not exist? Same with below in the template.

Reply:
Actually we have a guarantee that goal tags will exist here, think I re-used some logic from another form where tags was not strongly guaranteed so will just remove these checks entirely.